### PR TITLE
#731 Provide instructions to setup a multi-cluster env

### LINKED ISSUE #731: Describe how to externally run the helm-service and jmeter-service
With Keptn 0.8.0, we will have the possibility to run the helm-service and jmeter-service in a different cluster than the control plane. 
As a user, I would like to have documentation on how to set this up.

### PR DESCRIPTION
**Pull request by johannes-b**
Closes #731 
Signed-off-by: Johannes <johannes.braeuer@dynatrace.com>

Preview: https://deploy-preview-743--keptn.netlify.app/docs/0.8.x/operate/multi_cluster/

### CODE REVIEW

**Comment by christian-kreuzberger-dtx** [comment]
Hi,

I'm wondering if the steps described can be simplified a bit, as shown in my screenshot below:
image

```console
# download values.yaml from release branch
wget https://raw.githubusercontent.com/keptn/keptn/release-0.8.0-rc1/jmeter-service/chart/values.yaml
# edit values.yaml
nano values.yaml
# install jmeter-service from release assets with provided values.yaml
helm install jmeter-service https://github.com/keptn/keptn/releases/download/0.8.0-rc1/jmeter-service-0.8.0-rc1.tgz -n keptn-exec --create-namespace --values=values.yaml
# test jmeter-service
helm test jmeter-service -n keptn-exec
```

We could even go one step further and skip values.yaml, e.g.:

```
KEPTN_API_PROTOCOL="http"
KEPTN_API_TOKEN="xyz"
KEPTN_API_HOSTNAME="1.2.3.4"

# install jmeter-service from release assets
helm install jmeter-service https://github.com/keptn/keptn/releases/download/0.8.0-rc1/jmeter-service-0.8.0-rc1.tgz -n keptn-exec --create-namespace --values=remoteControlPlane.enabled=true,remoteControlPlane.api.token=$KEPTN_API_TOKEN,remoteControlPlane.api.hostname=$KEPTN_API_HOSTNAME,remoteControlPlane.api.protocol=$KEPTN_API_PROTOCOL
# test jmeter-service
helm test jmeter-service -n keptn-exec
```

IMHO it is enough to provide some examples (same as we do with the Advanced Keptn Install Options), and describe the values that can be set (see https://github.com/keptn/keptn/blob/release-0.8.0-rc1/helm-service/chart/README.md and https://github.com/keptn/keptn/blob/release-0.8.0-rc1/jmeter-service/chart/README.md)

**Comment by johannes-b** [comment]
@christian-kreuzberger-dtx many thanks for your feedback. I streamlined the installation of an exec plane service (according to your example.)

File: content/docs/0.8.x/operate/multi_cluster/index.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+{{< popup_image
+link="./assets/multi_cluster.png"
+caption="Multi-cluster setup"
+width="600px">}}
+
+* **Keptn Control plane**
+  * The control plane is the minimum set of components, which are required to run a Keptn and to manage projects, stages, and services, to handle events, and to provide integration points.
+  * The control plane orchestrates the task sequences defined in Shipyard, but does not actively execute the tasks.
+  * Minimum [Cluster size](./k8s_support/#control-plane)
```
Comment (christian-kreuzberger-dtx) [inline content/docs/0.8.x/operate/multi_cluster/index.md:19]: This link doesn't work for me. I think it should be `../k8s-support/#control-plane`

File: content/docs/0.8.x/operate/multi_cluster/index.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+* **Keptn Execution plane**
+  * The execution plane consists of all Keptn-services that are required to process all tasks (like deployment, test, etc.).
+  * The execution plane is the cluster where you deploy your application too and execute certain tasks of a task sequence. 
+  * Minimum [Cluster size](./k8s_support/#execution-plane)
```
Comment (christian-kreuzberger-dtx) [inline content/docs/0.8.x/operate/multi_cluster/index.md:24]: Same here

File: content/docs/0.8.x/operate/multi_cluster/index.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    keptn install --endpoint-service-type=[LoadBalancer, NodePort, ClusterIP]
+    ```
+
+## Install Keptn Execution plane
+
+In this release of Keptn, the execution plane services for deployment (`helm-service`) and testing (`jmeter-service`) can be installed via Helm Charts. 
+
+Please find the Helm Charts here: 
+
+  - `helm-service`: GitHub Release for [0.8.0](https://github.com/keptn/keptn/releases/tag/0.8.0) at **Assets** > `helm-service-0.8.0-dev.tgz`
```
Comment (christian-kreuzberger-dtx) [inline content/docs/0.8.x/operate/multi_cluster/index.md:72]: Filename is wrong: should be `helm-service-0.8.0.tgz` (or `helm-service-0.8.0-rc1.tgz` respectively). Same for jmeter-service.

File: content/docs/0.8.x/operate/multi_cluster/index.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+      * `production` - on Cluster-B  
+
+    To properly configure the execution plane services that run, for example, on **Cluster-A**, the distributor in the `values.yaml` needs to be configured:
+
+    ```
+    distributor:
+      projectFilter: ""                     # set the project, e.g., "sockshop" (to get events for the entire project)
+      stageFilter: ""                       # set the stage, e.g., "hardening" (to get events for the stage)
+      serviceFilter: ""                     # set the service, e.g., "carts" (to get events for the service )
+    ```
```
Comment (christian-kreuzberger-dtx) [inline content/docs/0.8.x/operate/multi_cluster/index.md:111]: In the above example, shouldn't the configuration for **Cluster-A** have `stageFilter` set to `hardening`?

File: content/docs/0.8.x/operate/multi_cluster/index.md [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    hostname: ""                        # < (3) set Keptn hostname (without /api)
+    apiValidateTls: true                # < (4 - optional) option to skip TLS verification
+    token: ""                           # < (5) set Keptn API token
+  ```
+
+  - Is `enabled` set to `true`?
+  - Is the Keptn API endpoint on `http` or `https`? 
+  - Is the hostname of the Keptn API endpoint correct, e.g. `my.keptn-dev.company.com` (without `/api`)
+  - Do you want to skip TLS verification?
+  - Is the Keptn API token correct. (You can find it in the Keptn Bridge)
```
Comment (christian-kreuzberger-dtx) [inline content/docs/0.8.x/operate/multi_cluster/index.md:218]: or by following the guide for authenticating `../install/#authenticate-keptn-cli`

File: content/docs/0.8.x/operate/multi_cluster/index.md [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+| ------------------------ | ----------------------- | -------------- |
+| `distributor.stageFilter` | Sets the stage this service belongs to | `""` |
+| `distributor.serviceFilter` | Sets the service this service belongs to | `""` |
+| `distributor.projectFilter` | Sets the project this service belongs to | `""` |
+| `remoteControlPlane.enabled` | Enables remote execution plane mode | `false` |
+| `remoteControlPlane.api.protocol` | Used protocol (http, https) | `"https"` |
+| `remoteControlPlane.api.hostname` | Hostname of the control plane cluster (and port) | `""` |
+| `remoteControlPlane.api.apiValidateTls` | Defines if the control plane certificate should be validated | `true` |
+| `remoteControlPlane.api.token` | Keptn API token | `""` |
+
```
Comment (christian-kreuzberger-dtx) [inline content/docs/0.8.x/operate/multi_cluster/index.md:159]: We should add an **uninstall** seciton aswell.

It's rather simple though:
```
helm uninstall jmeter-service -n keptn-exec
helm uninstall helm-service -n keptn-exec
```

File: content/docs/0.8.x/operate/multi_cluster/index.md (hunk 1 above)
Comment (johannes-b) [reply content/docs/0.8.x/operate/multi_cluster/index.md:19]: fixed it.

File: content/docs/0.8.x/operate/multi_cluster/index.md (hunk 2 above)
Comment (johannes-b) [reply content/docs/0.8.x/operate/multi_cluster/index.md:24]: fixed it.

File: content/docs/0.8.x/operate/multi_cluster/index.md [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+    caption="Installation options"
+    width="1000px">}}
+
+* Please make yourself familiar with the ways of exposing Keptn as explained [here](../install/#create-or-bring-a-kubernetes-cluster). Then come back and continue installing Keptn control plane.
+
+* To install the control plane, execute `keptn install` with the option you chose for exposing Keptn:
+
+    ```console
+    keptn install --endpoint-service-type=[LoadBalancer, NodePort, ClusterIP]
+    ```
```
[the remark this answers is not in the document]
Comment (johannes-b) [reply content/docs/0.8.x/operate/multi_cluster/index.md:64]: I reduced it to the max. Just the `keptn install` + links to retrieve endpoint/token.

File: content/docs/0.8.x/operate/multi_cluster/index.md (hunk 5 above)
Comment (johannes-b) [reply content/docs/0.8.x/operate/multi_cluster/index.md:218]: I added it.

File: content/docs/0.8.x/operate/multi_cluster/index.md (hunk 3 above)
Comment (johannes-b) [reply content/docs/0.8.x/operate/multi_cluster/index.md:72]: thx. fixed it

File: content/docs/0.8.x/operate/multi_cluster/index.md (hunk 4 above)
Comment (johannes-b) [reply content/docs/0.8.x/operate/multi_cluster/index.md:111]: that is true.

File: content/docs/0.8.x/operate/multi_cluster/index.md (hunk 6 above)
Comment (johannes-b) [reply content/docs/0.8.x/operate/multi_cluster/index.md:159]: I added the section.

**Comment by johannes-b** [comment]
@christian-kreuzberger-dtx again, many thanks for your feedback!
I adapted the docu accordingly.

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM - @johannes-b you can decide IF and when to merge it.

**Pull request merged**